torchci/clickhouse_queries/master_commit_red_percent/query.sql

```diff
@@ -80,24 +80,28 @@ any_red AS (
 classified_red AS (
     SELECT
         granularity_bucket,
-        ARRAY_CREATE(
-            ARRAY_CREATE('Broken trunk', AVG(broken_trunk_red)),
-            ARRAY_CREATE('Flaky', AVG(all_red) - AVG(broken_trunk_red)),
-            ARRAY_CREATE('Total', AVG(all_red))
-        ) AS metrics,
+        arrayJoin(
+            array(
+                array('Broken trunk', toString(AVG(broken_trunk_red))),
+                array(
+                    'Flaky',
+                    toString(AVG(all_red) - AVG(broken_trunk_red))
+                ),
+                array('Total', toString(AVG(all_red)))
+            )
+        ) AS metrics
```

Comment:
Curious as to why you cast to string for all of these?

Reply:
Oh, I learn that CH only accepts array of the same type, i.e. Array(String), while Rockset, like Python, allows different types to mix together (string and float).  We could have a TODO task here to see if this query could be rewritten in a better way.